src/os/bluestore/bluestore_dump.cc

```diff
@@ -0,0 +1,438 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <boost/program_options/variables_map.hpp>
+#include <boost/program_options/parsers.hpp>
+
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+#include <time.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include "global/global_init.h"
+#include "common/ceph_argparse.h"
+#include "include/stringify.h"
+#include "common/errno.h"
+#include "common/safe_io.h"
+
+#include "os/bluestore/BlueFS.h"
+#include "os/bluestore/BlueStore.h"
+#include "common/admin_socket.h"
+#include "common/url_escape.h"
+namespace po = boost::program_options;
+
+#include "bluestore_dump.h"
+
+void Dump::dump(const bluestore_blob_t& t)
```

Comment:
What's I don't like about this and below dump() func implementations is that they are set apart from corresponding  structs. May be define such dumps as members of relevant structs/classses?